manifests/prepare/package.pp

```diff
@@ -0,0 +1,43 @@
+# == Class puppet_agent::prepare::package
+#
+# The only job this class has is to ensure that the correct puppet-agent
+# package is downloaded locally for installation.  This is used on platforms
+# without package managers capable of working with a remote https repository.
+#
+# [package_file_name]
+#   The puppet-agent package file to retrieve from the master.
+#
+class puppet_agent::prepare::package(
+  $package_file_name,
+) {
+  assert_private()
+
+  # Guard this so that we do not perform expensive checksum logic on the master
+  # for the large puppet-agent file if we have already upgraded.
+  if $puppet_agent::params::master_agent_version != $::aio_agent_version {
+    $pe_server_version = pe_build_version()
+
+    if $::osfamily == 'windows' {
+      $tag = $::puppet_agent::arch ? {
+        'x64' => 'windows-x86_64',
+      }
+      $source = "puppet:///pe_packages/${pe_server_version}/${tag}/${package_file_name}"
+      $dest = "C:\\Program Files\\Puppet Labs\\Puppet Enterprise\\packages"
+      $file_dest = "C:\\Program Files\\Puppet Labs\\Puppet Enterprise\\packages\\${package_file_name}"
+    } else {
+      $source = "puppet:///pe_packages/${pe_server_version}/${::platform_tag}/${package_file_name}"
+      $dest = "/opt/puppetlabs/packages"
```

Comment:
Ya, this was on my list of things to clean up.  @highb what do you think about pulling this back into params, and then either plumbing it through alongside $package_file_name or referencing it as something like $pupppet_agent::params::local_packages_dir.  Either way, we can then replace all the /opt/puppetlabs/packages references.

And then I think the reference to puppet_agent::prepare::package::file_dest in install.pp just becomes this variable + package_file_name.